src/test/regress/expected/multi_router_planner.out

```diff
@@ -2041,6 +2041,81 @@ DEBUG:  Plan is router executable
 (6 rows)
 
 SET client_min_messages to 'NOTICE';
+-- test that a connection failure marks placements invalid
+SET citus.shard_replication_factor TO 2;
+CREATE TABLE failure_test (a int, b int);
+SELECT master_create_distributed_table('failure_test', 'a', 'hash');
+ master_create_distributed_table 
+---------------------------------
+ 
+(1 row)
+
+SELECT master_create_worker_shards('failure_test', 2);
+ master_create_worker_shards 
+-----------------------------
+ 
+(1 row)
+
+CREATE USER router_user;
+NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
+HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
+GRANT INSERT ON ALL TABLES IN SCHEMA public TO router_user;
+\c - - - :worker_1_port
+CREATE USER router_user;
+NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
+HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
+GRANT INSERT ON ALL TABLES IN SCHEMA public TO router_user;
+\c - router_user - :master_port
+-- first test that it is marked invalid inside a transaction block
+-- we will fail to connect to worker 2, since the user does not exist
+BEGIN;
+INSERT INTO failure_test VALUES (1, 1);
+WARNING:  connection error: localhost:57638
+DETAIL:  no connection to the server
+
+WARNING:  connection error: localhost:57638
+DETAIL:  no connection to the server
```

Comment:
Which kind of feels like a bug, if the `BEGIN` failed there's no way we want to keep sending commands!